### PR TITLE
LPD-38875 Truncate text values when max length exceeded

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-38875

Indexing DDM structures may fail when text values exceed the maximum length.  Treat them the same way as rich text fields, truncating the value before sending it to the indexer.

# How can you verify that it works?

This was breaking some of our tests. Run
```
$ cd asset-auto-tagger-opennlp-test && gw testIntegration --tests '*JournalArticleOpenNLPDocumentAssetAutoTaggerTest'
```

The tests will pass, but you should see failures in the logs.  After this fix, the failures no longer occur.

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java [hunk 1]
```diff
@@ -684,7 +684,9 @@ else if (value instanceof Object[]) {
 					document.addDate(name.concat("_date"), dateValues);
 				}
 			}
-			else if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT)) {
+			else if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT) ||
+					 type.equals(DDMFormFieldTypeConstants.TEXT)) {
+
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:689]: @adolfopa I'm not sure if calling `_htmlParser.extractText(valueString)` for text fields [here](https://github.com/liferay-content-management/liferay-portal/blob/07b31901b2131e9529ec8562f72cbfd20ad02129/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java#L696) could cause unintended consequences.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java [hunk 2]
```diff
@@ -684,7 +684,9 @@ else if (value instanceof Object[]) {
...
 				List<String> richTextValues = new ArrayList<>(
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:690]: @adolfopa I think we could rename it to `textValues` to make it more clear.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java [hunk 3]
```diff
@@ -695,14 +697,7 @@ else if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT)) {
...
 
-					if (richTextValue.length() >
-							_SORTABLE_TEXT_FIELDS_TRUNCATED_LENGTH) {
-
-						richTextValue = richTextValue.substring(
-							0, _SORTABLE_TEXT_FIELDS_TRUNCATED_LENGTH);
-					}
-
-					truncatedValues.add(richTextValue);
+					truncatedValues.add(_truncate(richTextValue));
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:700]: @adolfopa I think we could rename it to `textValue` to make it more clear.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 1 above)
Comment (adolfopa) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:689]: Makes sense. I'll split this in two different branches, and treat them independently.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 3 above)
Comment (adolfopa) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:698]: I've separated both cases in the conditional, so I'm keeping the names as is.

Comment (adolfopa) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:698]: I've separated these two cases in the conditional, so I believe the rename is no longer necessary.

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**